Platform/CommonBoardPkg/CfgData/CfgData_Common.yaml

```diff
@@ -26,6 +26,30 @@
                      Specify payload ID string. Empty will boot default payload. Otherwise, boot specified payload ID in multi-payload binary.
       length       : 0x04
       value        : ''
+  - PayloadSelGpio :
+    - $STRUCT      :
+        name         : GPIO pin for switching payload
+        struct       : PAYLOAD_SEL_GPIO_PIN
+        length       : 0x04
+        value        : 0
+    - PinNum       :

```

Comment:
Is this generic for all platforms ?    How can the end user determine the PinNum value ?
Which Group and which pin ?     
The old CFL allows you to select group and pin.

Reply:
Yes this is generic, and yes for now it seems like 'magic number' (just like APL and WHL), but user can refer to each platform GPIO lib (eg. Silicon\CoffeelakePkg\Include\GpioPinsCnlH.h) to check for the pin number .

Yes CFL did expose the option in CFL Sil cfg data, but it is not being used anywhere in code, hence it wont take effect at all if user configure any specific group.

What do you think?

Reply:
From user perspective, I think we should make it easy for user to select directly using group and pin number, ex: GPIO A10, D11, etc.  Magic number is hard for user to config.   SBL code can use the Group and Pin number to calculate the magic number required.